core/src/com/unciv/ui/worldscreen/bottombar/BattleTable.kt

```diff
@@ -263,16 +275,23 @@ class BattleTable(val worldScreen: WorldScreen): Table() {
 
         val canReach = attacker.unit.currentTile.getTilesInDistance(attacker.unit.getRange()).contains(targetTile)
 
-        if (!worldScreen.isPlayersTurn || !attacker.canAttack() || !canReach) {
+        if (!worldScreen.isPlayersTurn || !attacker.canAttack() || !canReach || !canNuke) {
             attackButton.disable()
             attackButton.label.color = Color.GRAY
         }
         else {
             attackButton.onClick {
                 try {
-                    Battle.nuke(attacker, targetTile)
-                    worldScreen.mapHolder.unitActionOverlay?.remove() // the overlay was one of attacking
-                    worldScreen.shouldUpdate = true
+                    if(!(targetTile.getCity()?.civInfo?.getDiplomacyManager(attacker.getCivInfo())?.canDeclareWar() ?:true))
```

Comment:
Didn't we agree that this code with notification logic should be removed?
Currently, if you are trying to nuke your own territory it causes the application crash 💥 
However, you should be able to nuke your own tiles (e.g. with enemies on it).

Reply:
Agree, this is still not easily readable :)
I know I go on about it a lot, but that's one of the main points of open source code - it has to be easily legible for whoever may take an interest in it :)

Reply:
yeah this was a mistake lol! like I long deleted this line, but I apparently made another (local) change on my laptop and forgot about it(prior to that I worked on my pc)! So this line shouldnt have been in the commit in the first place.

Reply:
No rush, there's no deadline :D
Take all the time you need